lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp

```diff
@@ -21,7 +21,6 @@
 #include "llvm/ADT/TypeSwitch.h"
 #include "llvm/Support/MathExtras.h"
 
-using namespace mlir;
 using namespace circt;
 using namespace circt::handshake;
 using namespace circt::firrtl;
```

Comment:
It might make sense to leave the 'using namespace mlir' in this .cpp file.  They aren't harmful there, and that would remove a bunch of clutter.

Reply:
Yeah, this points to the problem I was describing in the message where `using namespace mlir;` creates an ambiguous lookup by name, even though everything is aliased to the same thing:
```
HandshakeToFIRRTL.cpp:952:3:
error: reference to 'DenseMap' is ambiguous
  DenseMap<size_t, Value> argIndexValues;
note: candidate found by name lookup is 'mlir::DenseMap'
  using DenseMap = llvm::DenseMap<KeyT, ValueT, KeyInfoT, BucketT>;
note: candidate found by name lookup is 'circt::DenseMap'
  using DenseMap = llvm::DenseMap<KeyT, ValueT, KeyInfoT, BucketT>;
 ```
I'm not sure what to do about this one yet, but I agree that we should be able to use `using namespace mlir;` when we want to without this explosion.

Reply:
Ah, I just noticed it is specific to the way we import templated types.  Non-templated classes work fine.
```
template <typename ValueT, typename ValueInfoT = DenseMapInfo<ValueT>>
using DenseSet = llvm::DenseSet<ValueT, ValueInfoT>;
```

Reply:
I think I figured out how to solve this problem, the details are in the code https://github.com/llvm/circt/pull/586/files#diff-773c48b0ebb0d02e5f73e3221a0dd9db52ea7dc0701c2d238f21d632016434a3R28.  The header file is a lot simpler overall 😄